crates/ironrdp-async/src/connector.rs

```diff
@@ -68,6 +84,87 @@ where
     Ok(result)
 }
 
+async fn resolve_generator(
+    generator: &mut CredSspProcessGenerator<'_>,
+    network_client: &mut impl AsyncNetworkClient,
+) -> ConnectorResult<ClientState> {
+    let mut state = generator.start();
+    loop {
+        match state {
+            GeneratorState::Suspended(request) => {
+                let response = network_client.send(&request).await?;
+                state = generator.resume(Ok(response));
+            }
+            GeneratorState::Completed(client_state) => {
+                break Ok(client_state.map_err(|e| custom_err!("cannot resolve generator state", e))?)
+            }
+        }
+    }
+}
+
+#[instrument(level = "trace", skip(network_client, framed, buf, server_name, server_public_key))]
+async fn perform_credssp_step<S>(
+    framed: &mut Framed<S>,
+    connector: &mut ClientConnector,
+    buf: &mut WriteBuf,
+    server_name: impl Into<ServerName>,
+    server_public_key: Vec<u8>,
+    mut network_client: Option<impl AsyncNetworkClient>,
+) -> ConnectorResult<ironrdp_connector::Written>
```

Comment:
For the same reason.
```suggestion
) -> ConnectorResult<()>
```